### ISSUE TITLE
adbExec forward --list fails with android test distribution 

### ISSUE THREAD

**Issue by 99heitor** [saikrishna321/AppiumTestDistribution#256]
The following error happens very often when executing Android tests, which causes all steps to skip:

```
[debug] [ADB] Running '/home/heitor/Android/Sdk/platform-tools/adb' with args: ["-P",5037,"-s","59a1181b","forward","--list"]
[Chromedriver] Error: Error executing adbExec. Original error: 'Command '/home/heitor/Android/Sdk/platform-tools/adb -P 5037 -s 59a1181b forward --list' exited with code 1'; Stderr: 'error: protocol fault (couldn't read status message): Success'; Code: '1'
    at ADB.execFunc$ (../../../lib/tools/system-calls.js:317:13)
    at tryCatch (/home/heitor/.npm-global/lib/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:67:40)
    at GeneratorFunctionPrototype.invoke [as _invoke] (/home/heitor/.npm-global/lib/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:315:22)
    at GeneratorFunctionPrototype.prototype.(anonymous function) [as throw] (/home/heitor/.npm-global/lib/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:100:21)
    at GeneratorFunctionPrototype.invoke (/home/heitor/.npm-global/lib/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:136:37)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)
```

Also, when all steps are skipped the report shows "Pass".

I noticed @saikrishna321 [raised a related issue with the Appium project here](https://github.com/appium/appium/issues/9392). Did you manage to solve it?

**Comment by saikrishna321** [comment]
follow that thread, it has the solution ... I updated developer tools and that fixed

**Comment by 99heitor** [comment]
Unfortunately I had already tried updating developer tools. Seems like it wasn't enough in my case.

It usually happens after a lot of scenarios are executed (30+), and keeps on happening for the rest of the session. It's probably just adb's flakiness, and there's not much you could do. 
Feel free to close the issue if you want, I'll update if I find a solution.

**Comment by 99heitor** [comment]
I found out what my problem. I need some custom capabilities in my application that create a new adb port forward with every scenario. I was not removing these forwards when the scenario finished, so they stacked up until adb couldn't list them anymore running `adb forward --list`, causing the protocol fault I reported. Simply deleting the port forwards between scenarios using the @After tag on hooks solved the issue for me.
